musileko/client/gl/texture.zig

```diff
@@ -191,21 +213,33 @@ pub const PixelFormat = struct {
 };
 
 pub const PixelChannels = enum(c_uint) {
-    r = c.GL_RED,
-    rg = c.GL_RG,
-    rgb = c.GL_RGB,
-    rgba = c.GL_RGBA,
-    depth = c.GL_DEPTH_COMPONENT,
+    r,
+    rg,
+    rgb,
+    srgb,
+    rgba,
+    srgb_alpha,
+    depth,
 
     const Self = @This();
 
-    pub fn fromDimensions(dims: u32) Self {
-        return switch (dims) {
-            1 => .r,
-            2 => .rg,
-            3 => .rgb,
-            4 => .rgba,
-            else => .rgba,
+    pub fn fromDimensions(dims: u32, srgb: bool) Self {
```

Comment:
This could be changed so that srgb and non-srgb are in separate fns rather than giving this one an additional bool, because the majority of the time you want srgb textures from disk (for example png's are intended to be used as srgb data by default) and non-srgb textures for data (every g-buffer other than color) but a bool was a quick fix